src/sorted_vector.jl

```diff
@@ -0,0 +1,52 @@
+"""
+A `SortedVector` behaves like a standard Julia `Vector`, except that its elements are stored in sorted order, with an optional function `by` that determines the sorting order in the same way as `Base.searchsorted`.
+"""
+immutable SortedVector{T, F<:Function}
+    data::Vector{T}
```

Comment:
If I understand correctly, there will be a big performance hit for making the inner container an `AbstractVector`. The compiler won't be able to ascertain the actual type of the inner container at compile-time so there will be a lot more run-time dispatching.  To avoid this problem, you would need to make both the data type and the inner container type parameters of `SortedVector`.